File: k6-benchmark/go/main.go
package main

import (
	"log"
	"net/http"
)

const PORT = ":1800"

func TestHandler(w http.ResponseWriter, r *http.Request) {
	w.Write([]byte("Hello world"))
}

func main() {
	http.HandleFunc("/", TestHandler)
	log.Println("Go server serving at port", PORT)
	http.ListenAndServe(PORT, nil)
}

File: k6-benchmark/lua/app/main.lua
return require('luvit')(function (...)
  local http = require('http')

  http.createServer(function (req, res)
    local body = "Hello world\n"
    res:setHeader("Content-Type", "text/plain")
    res:setHeader("Content-Length", #body)
    res:finish(body)
  end):listen(1337, '0.0.0.0')
  
  print('Server running at port: 1337/')
end, ...)
